src/pages/HomePage/index.js

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { Link } from "react-router-dom";
+import SidebarLeft from "../../components/SidebarLeft";
+import SidebarRight from "../../components/SidebarRight";
+import {
+  setBooksCollections,
+  fetchFullData,
+} from "../../store/homepage/actions";
+import { selectBooksCollections } from "../../store/homepage/selectors";
+import "./index.css";
+
+export default function HomePage() {
+  const dispatch = useDispatch();
+  const booksCollections = useSelector(selectBooksCollections);
+
+  useEffect(() => {
+    dispatch(fetchFullData());
+  }, [dispatch, setBooksCollections]);
```

Comment:
in this case it shouldn't affect in any way, but careful on passing extra stuff to your useEffects because if its a variable you pass and it changes the `useEffect` will fire again